engine/src/main/java/org/terasology/rendering/dag/stateChanges/SetViewportToSizeOf.java

```diff
@@ -80,20 +89,24 @@ public boolean isTheDefaultInstance() {
 
     @Override
     public void update() {
-        if (defaultDynamicFBO == null) {
-            FBO fbo = frameBuffersManager.get(fboName);
-            task.setDimensions(fbo.width(), fbo.height());
-        } else {
-            task.setDimensions(defaultDynamicFBO.width(), defaultDynamicFBO.height());
-        }
+        task.setDimensions(getWidth(), getHeight());
     }
 
     @Override
     public String toString() { // TODO: used for logging purposes at the moment, investigate different methods
         return String.format("%30s: %s", this.getClass().getSimpleName(), fboName);
     }
 
-    public ResourceUrn getFboName() {
-        return fboName;
+    public int getWidth() {
+        return getFbo().width();
+    }
+
+    public int getHeight() {
+        return getFbo().height();
+    }
```

Comment:
It was a conscious decision to _not_ provide much in the way of accessors in all StateChange classes. 

It was tempting but I came to the conclusion that all that matters in the Intermediate List (listing nodes and state changes rather than tasks) is where that information comes from. In this class the width and height come from an FBO and that's why the toString method publishes the FBO's name to the log. The associated task (if it makes it into the task list) then publishes the deeper information, in this case the width and height.

So I'd prefer to remove these methods and use getFbo().width() straight into the update() method. 

Good idea about adding the getFbo() method though.